### PR TITLE
feat: Simple randomness beacon

### PR DESCRIPTION
**Pull request by SkidanovAlex**
This implements a simple biasable randomness beacon. The work on the
unbiasable randomness beacon is here:
https://github.com/nearprotocol/nearcore/commits/rnd_integration

The biasable randomness beacon in this commit just uses the VRF created
by the current block proposer, thus it has 1 bit of influence (the block
proposer can choose not to create a block), which is sufficient for the
block producers rotation, but should be used with caution for other use
cases.

Test plan
---------
No new tests added as part of the change.

### CODE REVIEW

File: near/src/runtime.rs [hunk 1]
```diff
@@ -399,6 +399,27 @@ impl RuntimeAdapter for NightshadeRuntime {
         Ok(())
     }
 
+    fn verify_block_vrf(
+        &self,
+        epoch_id: &EpochId,
+        block_height: BlockHeight,
+        prev_random_value: CryptoHash,
```
Comment (bowenwang1996) [inline near/src/runtime.rs]: better to pass as reference

**Comment by SkidanovAlex** [comment]
Yes, we need to make sure the staking transaction always contains ED25519 that is convertable to restretto. Will add.

File: core/primitives/src/errors.rs [hunk 2]
```diff
@@ -328,6 +328,8 @@ pub enum ActionErrorKind {
         #[serde(with = "u128_dec_format")]
         balance: Balance,
     },
+    /// An attempt to stake with a key that is not convertable to ristretto
+    UnsuitableStakingKey,
```
Comment (nearmax) [inline core/primitives/src/errors.rs]:
```suggestion
    UnsuitableStakingKey {public_key: PublicKey},
```

File: core/primitives/src/errors.rs [hunk 3]
```diff
@@ -592,6 +594,9 @@ impl Display for ActionErrorKind {
                 "Account {:?} tries to stake {}, but has staked {} and only has {}",
                 account_id, stake, locked, balance
             ),
+            ActionErrorKind::UnsuitableStakingKey { .. } => {
+                write!(f, "The staking key must be ED25519.")
```
Comment (nearmax) [inline core/primitives/src/errors.rs]: Please use the key in the error message.

File: core/primitives/src/block.rs [hunk 4]
```diff
@@ -387,6 +394,10 @@ pub struct Block {
     pub header: BlockHeader,
     pub chunks: Vec<ShardChunkHeader>,
     pub challenges: Challenges,
+
+    // Data to confirm the correctness of randomness beacon output
+    pub vrf_value: near_crypto::vrf::Value,
+    pub vrf_proof: near_crypto::vrf::Proof,
```
Comment (abacabadabacaba) [inline core/primitives/src/block.rs]: Why there are two fields `vrf_value` and `random_value` in different places?

Comment (SkidanovAlex) [reply core/primitives/src/block.rs]: The logic is the following: with this randomness beacon the `random_value = hash(vrf_value)`.
However, once we switch to the new randomness beacon, that will change to `random_value = new_beacon(hash(vrf_value))`. In other words, `vrf_value` will always stay the output of the VRF by the current block producer, while the way the `random_value` is computed can change.

**Pull request merged**